lib/routes/cab/routes.js

```diff
@@ -0,0 +1,149 @@
+import Joi from '@hapi/joi';
+import {
+    numberSchema,
+    stringSchema,
+    stringSchemaOptional
+} from 'utils/validationUtils';
+import {
+    createOneCab,
+    findAllCabs,
+    findOneCab,
+    updateOneCab
+} from 'daos/cabDao';
+import { findOneDriver } from 'daos/driverDao';
+import { sequelize } from 'models';
+
+module.exports = [
+    {
+        method: 'POST',
+        path: '/',
+        options: {
+            description: 'Create a new Cab',
+            notes: 'Creates a new Cab',
+            tags: ['api', 'cabs'],
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    driver_id: numberSchema,
+                    cab_type: stringSchema,
+                    cab_number: stringSchema
+                })
+            }
+        },
+        handler: async (request, h) => {
+            const { driverId, cabType, cabNumber } = request.payload;
+
+            // Check if the driver exists in the database
+            const driverExists = await findOneDriver(driverId);
+            if (!driverExists) {
+                return h.response({
+                    errors: [{ message: 'This driver does not exist.' }]
+                });
+            }
+
+            const newCab = await createOneCab({
+                driverId,
+                cabNumber,
+                cabType
+            });
+
+            return h.response({ cab: newCab });
+        }
+    },
+    {
+        method: 'POST',
+        path: '/nearby',
+        options: {
+            description: 'Get all cabs nearby provided location',
+            notes: 'Find all cabs',
+            tags: ['api', 'cabs'],
+            auth: false,
+            validate: {
+                payload: Joi.object({
+                    current_location: stringSchema,
+                    radius: numberSchema
+                })
+            }
+        },
+        handler: async (request, h) => {
+            const { currentLocation, radius } = request.payload;
+            const parsedCurrentLocation = JSON.parse(currentLocation);
+
+            const nearby = await sequelize.query(
+                `
+                        SELECT 
+                            first_name, 
+                            ST_Distance_Sphere(current_location, ST_GeomFromText('POINT(:lng :lat)', 4326)) AS "distance_m"
+                        FROM drivers
+                        HAVING distance_m <= :rad
+                    `,
+                {
+                    replacements: {
+                        lat: parsedCurrentLocation[0],
+                        lng: parsedCurrentLocation[1],
+                        rad: radius
+                    }
+                }
+            );
+
+            return h.response({ cabsNearby: nearby[0] });
```

Comment:
Why return one car? We need all nearby cars right?

Reply:
Yeah as mentioned that the response I get from the raw query is something like this
```json
{
    "nearby": [ /* all the cabs */ ],
    "nearby": [ /* all the cabs */ ]
}
```
Therefore to send just once I have done `nearby[0]`